integr8ly/integreatly-operator
# MGDAPI-5887 - Create go module for 3scale apis to import to integreatly-opeartor

# Issue link
https://issues.redhat.com/browse/MGDAPI-5887

### This PR will be merged after 3scale PR: https://github.com/3scale/3scale-operator/pull/854
**PR require update of go.mod - replace private repo reference to 3scale-operator**: `github.com/valerymo/3scale-operator v0.0.0-20230831081915-2a8db134a385`

# What
Reduced the number of 3-scale dependencies in Roam.
This was achieved through a 3scale refactoring and reducing APIs dependencies on PKG packages in 3scale. The result is the same as we checked for API module, so final decision was Not to create separate apis module.

# Verification steps
- get current branch
- check the number of dependencies - count of files in vendors dir, compare with master:
```
$ pwd
.../integreatly-operator/vendor
[vendor] (master)$ find . -type f |wc -l
7001
[vendor] $ find . -type f |wc -l
6770
$ date
Thu Aug 31 12:49:27 IDT 2023
```
- Install Rhoam locally, and check that installation is successfull
```
[integreatly-operator] (MGDAPI-5887)$ oc get rhmi rhoam -n redhat-rhoam-operator -o json | jq -r '.status.stage'
complete
```
